ocs_ci/ocs/container.py

```diff
@@ -0,0 +1,250 @@
+        ] = self.secret_data_name
+        self.sc_data['parameters'][
+            'csi.storage.k8s.io/provisioner-secret-namespace'
+        ] = defaults.ROOK_CLUSTER_NAMESPACE
+
+        self.sc_data['parameters']['clusterID'] = defaults.ROOK_CLUSTER_NAMESPACE
+        self.sc_data['reclaimPolicy'] = self.reclaim_policy
+
+    def _setup_pvc(self):
+        """
+         Setup PVC data
+        """
+        self.pvc_data = templating.load_yaml_to_dict(constants.CSI_PVC_YAML)
+        if self.pvc_name is None:
+            self.pvc_name = create_unique_resource_name('test', 'pvc')
+        self.pvc_data['metadata']['name'] = self.pvc_name
+        self.pvc_data['metadata']['namespace'] = self.namespace
+        self.pvc_data['spec']['accessModes'] = [self.access_mode]
+        self.pvc_data['spec']['storageClassName'] = self.sc_name
+        if self.size:
+            self.pvc_data['spec']['resources']['requests']['storage'] = self.size
+
+    def _create_rbd_pvc(self):
+        """
+         Create a PVC backed by RBD
+        """
+        self.rbd_secret_obj = self._create_ocs(self.secret_data)
+        self.ceph_block_pool_obj = self._create_ocs(self.block_data)
+        self.rbd_sc_obj = self._create_ocs(self.sc_data)
+        self.pvc_obj = self._create_ocs(self.pvc_data, constants.STATUS_BOUND)
+
+    def _create_fs_pvc(self):
+        """
+         Create a PVC backed by CephFS
+        """
+        self.cephfs_secret_obj = self._create_ocs(self.secret_data)
+        self.fs_sc_obj = self._create_ocs(self.sc_data)
+        self.pvc_obj = self._create_ocs(self.pvc_data, constants.STATUS_BOUND)
+
+    def _delete_rbd_pvc(self):
+        """
+         delete rbd pvc
+        """
+        self._delete_ocs(self.pvc_obj)
+        self._delete_ocs(self.rbd_sc_obj)
+        self._delete_ocs(self.ceph_block_pool_obj)
+        self._delete_ocs(self.rbd_secret_obj)
+
+    def _delete_fs_pvc(self):
+        """
+         Delete fs pvc
+        """
+        self._delete_ocs(self.pvc_obj)
+        self._delete_ocs(self.fs_sc_obj)
+        self._delete_ocs(self.cephfs_secret_obj)
+
+    def _create_ocs(self, data, state=None):
+        obj = OCS(**data)
+        obj.create(do_reload=True)
+        # add label so that we can delete them with ease
+        obj.ocp.add_label(obj.name, constants.OCS4_TEST_LABEL)
+        if self.wait and state is not None:
+            obj.ocp.wait_for_resource(
+                condition=state,
+                resource_name=obj.name,
+                timeout=60
+            )
+        return obj
+
+    def _delete_ocs(self, obj):
+        obj.delete()
+        obj.ocp.wait_for_delete(obj.name)
+
+    def create(self):
+        """
+         create pvc and all required dependency objects
+        """
+        if self.interface_type == constants.CEPHBLOCKPOOL:
+            self._create_rbd_pvc()
+        else:
+            self._create_fs_pvc()
+
+    def delete(self):
```

Comment:
shouldn't it be called delete_pvc()?

Reply:
Since the object itself is a contained PVC, the delete implies that and better not to add additional suffix.